supported/osg-htc/osg-hosted-ce/templates/configmap.yaml

```
@@ -130,6 +130,34 @@ data:
     JOB_ROUTER_ENTRIES = [Name = "Hosted_CE_default_route"]
 
     JOB_ROUTER_ROUTE_NAMES = Hosted_CE_default_route
+
+    # New Job Router transforms for token and GridResource construction
+    JOB_ROUTER_TRANSFORM_Token @=jrt
+        SET TransferInput "$(MY.TransferInput),/usr/share/condor-ce/glidein-tokens/$(MY.Owner)/ce_$(MY.Owner).idtoken"
+    @jrt
+    JOB_ROUTER_POST_ROUTE_TRANSFORM_NAMES = $(JOB_ROUTER_POST_ROUTE_TRANSFORM_NAMES) Token
+
+    JOB_ROUTER_TRANSFORM_GridResource @=jrt
+        grid_resource = batch {{ .Values.RemoteCluster.Batch }} $(MY.Owner)@{{ .Values.RemoteCluster.LoginHost }}
+        grid_resource = $(grid_resource) --rgahp-glite ~/{{ .Values.RemoteCluster.BoscoDir | default "~/bosco"}}/glite
+        {{ if .Values.BoscoOverrides.TarballURL | default "" | contains "bosco-1.3"  }}
+        grid_resource = $(grid_resource) --rgahp-script batch_gahp
+        {{ end }}
+        {{ if not .Values.RemoteCluster.SSHBatchMode }}
+        grid_resource = $(grid_resource) --rgahp-nobatchmode
+        {{ end }}
+        {{ if not .Values.RemoteCluster.LoginShell }}
+        grid_resource = $(grid_resource) --rgahp-nologin
+        {{ end }}
+    @jrt
+    JOB_ROUTER_PRE_ROUTE_TRANSFORM_NAMES = $(JOB_ROUTER_PRE_ROUTE_TRANSFORM_NAMES) GridResource
+
+    # To use the new Job Router syntax, uncomment these lines:
+    #JOB_ROUTER_USE_DEPRECATED_ROUTER_ENTRIES = False
+    #JOB_ROUTER_ROUTE_Hosted_CE_default_route @=jrt
+    #    GridResource = "$(grid_resource)"
+    #@jrt
+
```

```suggestion
    {{ if .Values.HTCondorCE.JobRouterTransforms }}
    # Use the new job router syntax
    JOB_ROUTER_USE_DEPRECATED_ROUTER_ENTRIES = False
    JOB_ROUTER_ROUTE_Hosted_CE_default_route @=jrt
        GridResource = "$(grid_resource)"
    @jrt
    {{ end }}
```

Since operators (@ashtongraves @coldbeer101 @jdost321 ) should avoid directly modifying CE or remote blah directly, we should add support for this through a config flag in https://github.com/osg-htc/helm-charts/blob/main/supported/osg-htc/osg-hosted-ce/values.yaml. The above suggestion would look like the following where we would sensibly default to `false`:

```
HTCondorCE:
  JobRouterTransforms: false
```

I think the above is probably not what we want to use. Maybe something like

```
JobRouter:
  UseTransforms: false
```

But I'd like the operators to provide input on how they'd like to apply this config.

---

I like this idea.

---

The interface could also just be at the top level:

```
JobRouterUseTransforms: false
```

---

I prefer it done that way as well

---

Any other opinions on how the 'use new router syntax' value should be named?

---

I have no preference on the naming. In my opinion this should just be temporary until we get all CEs on the new syntax than get rid of having it set via values and just have it set by default.

---

Agreed.

---

As for the section it goes in, I don't think it's worth creating a separate section for this unless we have immediate plans to add more things for a JobRouter section. Again I'm hoping this is temporary until we get everything on the new syntax (which from our hosted ce meetings sounds like priority right now). So I say just make it a standalone value with the intention for it to be removed eventually